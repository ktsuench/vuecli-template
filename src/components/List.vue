<template>
  <ul>
    <li v-for="item in items" :key="item">{{ item }}</li>
  </ul>
</template>

<script>
export default {
  name: 'list',
  props: {
    items: {
      type: Array,
      default: () => [],
      validator: prop => prop.every(item => typeof item === 'string'),
    },
  },
};
</script>
