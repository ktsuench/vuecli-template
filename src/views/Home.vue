<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png" />
    <message-toggle />
    <list :items="['list item 1', 'list item 2', 'list item 3']" />
  </div>
</template>

<script>
// @ is an alias to /src
import MessageToggle from '@/components/MessageToggle.vue';
import List from '@/components/List.vue';
export default {
  name: 'home',
  components: {
    MessageToggle,
    List,
  },
};
</script>
