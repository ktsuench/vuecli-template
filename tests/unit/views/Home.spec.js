import { shallowMount } from '@vue/test-utils';
import Home from '@/views/Home.vue';
import List from '@/components/List.vue';
import MessageToggle from '@/components/MessageToggle.vue';

describe('Home.vue', () => {
  describe('data.listItems property', () => {
    const wrapper = shallowMount(Home);
    const listItemsData = wrapper.vm.$data.listItems;

    it('is array of strings', () => {
      expect(listItemsData).toStrictEqual([
        'list item 1',
        'list item 2',
        'list item 3',
      ]);
    });
  });

  it('renders view', () => {
    const wrapper = shallowMount(Home);
    expect(wrapper.contains(List));
    expect(wrapper.contains(MessageToggle));
  });

  it('matches snapshot', () => {
    const wrapper = shallowMount(Home);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
