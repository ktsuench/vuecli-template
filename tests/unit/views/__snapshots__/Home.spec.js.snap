// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue matches snapshot 1`] = `
<div class="home"><img alt="Vue logo" src="../assets/logo.png">
  <message-toggle-stub></message-toggle-stub>
  <list-stub items="list item 1,list item 2,list item 3"></list-stub>
</div>
`;
